public/app/plugins/datasource/loki/queryUtils.ts

```diff
@@ -293,3 +297,39 @@ export const getLokiQueryFromDataQuery = (query?: DataQuery): LokiQuery | undefi
 
   return query;
 };
+
+export function formatLogqlQuery(query: string, datasource: LokiDatasource) {
```

Comment:
Before merging, we need a couple of test cases here, specially for the transformation part which deals with variables.